### PR TITLE
Autocomplete demo with custom template

### LINKED ISSUE #2505: mdAutocomplete: provide another demo using more complex templates
The mdContactChip directive uses autoComplete with a nice template; but this is rather hidden from public view. It would be nice to demonstrate how mdAutoComplete supports more sophisticated templates for **cool** list rendering.

### PR DESCRIPTION
**Pull request by marcysutton**
As requested, the Autocomplete demo now shows how to use custom templates for dropdown list items (see "Custom Template" listed second in mdAutocomplete demo). To work around our docs prepending CSS classes onto demo `style.css`, I copied the custom styles into our docs stylesheet.

Closes #2505

### CODE REVIEW

**Comment by marcysutton** [comment]
Note: need to copy icon to asset cache

**Comment by marcysutton** [comment]
@ThomasBurleson note added to the docs stylesheet about why styles are duplicated.

File: docs/app/img/icons/octicon-repo.svg [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<svg xmlns="http://www.w3.org/2000/svg" x="0px" y="0px" viewBox="0 0 48 48">
```
Comment (matthewrfindley) [inline docs/app/img/icons/octicon-repo.svg]: You will need to generate the asset cache for this example to work in codepen.  https://github.com/angular/material/blob/master/docs/guides/CODEPEN.md

File: src/components/autocomplete/demoCustomTemplate/img/icons/octicon-repo.svg [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<svg xmlns="http://www.w3.org/2000/svg" x="0px" y="0px" viewBox="0 0 48 48">
```
Comment (matthewrfindley) [inline src/components/autocomplete/demoCustomTemplate/img/icons/octicon-repo.svg]: See above comment about asset cache.

File: docs/app/img/icons/octicon-repo.svg (hunk 1 above)
Comment (ThomasBurleson) [inline docs/app/img/icons/octicon-repo.svg]: Done.
@matthewrfindley  - those docs are so very helpful. #awesome.

**Pull request closed without merging**